src/custom-elements/src/components/date/date.tsx

```diff
@@ -0,0 +1,269 @@
+import {
+  Component,
+  Event,
+  EventEmitter,
+  Method,
+  Prop,
+  State,
+  Watch
+} from '@stencil/core';
+import dayjs, { Dayjs } from 'dayjs';
+import { CLASSES } from '../../constants/constants';
+
+const WEEK_CLASS_PART = ['sun', 'mon', 'tue', 'wed', 'thu', 'fri', 'sat'];
+
+@Component({
+  tag: 'chi-date',
+  styleUrl: 'date.scss',
+  scoped: true
+})
+export class Date {
+  /**
+   * Selected date in the date picker
+   */
+  @Prop({ reflectToAttr: true, mutable: true }) value: string;
+
+  /**
+   * Locale to use in date picker
+   */
+  @Prop({ reflectToAttr: true }) locale = 'en';
+
+  /**
+   * Minimum eligible date
+   */
+  @Prop({ reflectToAttr: true }) min = '01/01/1900';
+
+  /**
+   * Maximum eligible date
+   */
+  @Prop({ reflectToAttr: true }) max = '12/31/2099';
+
+  /**
+   * Date format used in the attributes and how it will be shown to the user.
+   */
+  @Prop({ reflectToAttr: true }) format = 'MM/DD/YYYY';
+
+  @State() viewMonth = dayjs();
+
+  @Watch('value')
+  dateChanged(newValue: string, oldValue: string) {
+    if (newValue !== oldValue) {
+      this._vm.date = newValue ? dayjs(newValue) : null;
+      if (this._vm.date && !this._vm.date.isValid()) {
+        throw new Error(`Date ${newValue} has an invalid format. `);
+      }
+      this.viewMonth = this._vm.date;
+    }
+  }
+
+  @Watch('locale')
+  localeChanged() {
+    this._initCalendarViewModel();
+  }
+
+  @Watch('viewMonth')
+  viewMonthChanged(newValue: Dayjs, oldValue: Dayjs) {
+    if (
+      newValue.month() !== oldValue.month() ||
+      newValue.year() !== oldValue.year()
+    ) {
+      this._updateViewMonth();
+    }
+  }
+
+  @Watch('min')
+  minChanged(newValue: string) {
+    const newMinDate = this.fromString(newValue);
+    if (!newMinDate.isValid()) {
+      throw new Error(`Min date value ${newValue} has an invalid format. `);
+    }
+    this._vm.min = newMinDate;
+  }
+
+  @Watch('max')
+  maxChanged(newValue: string) {
+    const newMaxDate = this.fromString(newValue);
+    if (!newMaxDate.isValid()) {
+      throw new Error(`Max date value ${newValue} has an invalid format. `);
+    }
+    this._vm.max = newMaxDate;
+  }
+
+  /**
+   * Sets date
+   */
+  @Method()
+  setDate(date) {
+    this.value = date;
+  }
+
+  /**
+   * Gets date
+   */
+  @Method()
+  getDate() {
+    return Promise.resolve(this.value);
+  }
+
+  /**
+   * Date change value event
+   */
+  @Event({ eventName: 'chiDateChange' }) eventChange: EventEmitter;
+
+  _vm: {
+    date: Dayjs;
+    today: Dayjs;
+    weekStartClass: string;
+    monthStartClass: string;
+    weekDays: Dayjs[];
+    monthDays: Dayjs[];
+    min: Dayjs;
+    max: Dayjs;
+  };
+
+  private _initCalendarViewModel(): void {
+    dayjs.locale(this.locale);
```

Comment:
Please, always try to define consts and lets at the beginning of functions (when possible)

Reply:
You're right, but `Dayjs.locale` function has to be called before any other Dayjs call, so it is not possible here.